### PR TITLE
Centrifuge App: Tinlake invest

### PR DESCRIPTION
**Pull request by onnovisser**
### Description

Tinlake investing

### Approvals

- [ ] Dev

### Screenshots
Screenshot 2023-02-23 at 14 20 14
Screenshot 2023-02-22 at 18 06 01
Screenshot 2023-02-22 at 18 06 11
Screenshot 2023-02-22 at 18 06 24
Screenshot 2023-02-22 at 18 10 58

### CODE REVIEW

File: centrifuge-app/src/utils/tinlake/useTinlakePools.ts [hunk 1]
```diff
@@ -188,11 +192,11 @@ async function getPools(pools: IpfsPools): Promise<{ pools: TinlakePool[] }> {
...
-      {
-        target: pool.addresses.FEED,
-        call: ['currentNAV()(uint256)'],
-        returns: [[`${poolId}.netAssetValue`, toCurrencyBalance]],
-      },
+      // {
+      //   target: pool.addresses.FEED,
+      //   call: ['currentNAV()(uint256)'],
+      //   returns: [[`${poolId}.netAssetValue`, toCurrencyBalance]],
+      // },
```
Comment (Hornebom) [inline centrifuge-app/src/utils/tinlake/useTinlakePools.ts:199]: Are these lines still needed? If so, how do you feel about saving them in an issue?

File: centrifuge-app/src/utils/tinlake/useTinlakePools.ts [hunk 2]
```diff
@@ -527,20 +481,20 @@ async function getPools(pools: IpfsPools): Promise<{ pools: TinlakePool[] }> {
...
         total: data.reserve,
       },
       nav: {
-        latest: data.netAssetValue,
+        latest: data.netAssetValue ?? new CurrencyBalance(0, 18),
         lastUpdated: new Date().toISOString(),
       },
       createdAt: null,
       isInitialised: true,
       // loanCollectionId: string | null;
```
Comment (Hornebom) [inline centrifuge-app/src/utils/tinlake/useTinlakePools.ts:496]: Can this be deleted?

**Review by Hornebom — APPROVED** [review]
Remove empty file: `centrifuge-js/src/modules/tinlake/lender.ts`

File: centrifuge-app/src/utils/tinlake/useTInlakeInvestment.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    // {
+    //   target: pool.addresses.SENIOR_TOKEN,
+    //   call: ['balanceOf(address)(uint256)', address],
+    //   returns: [['senior.balance', toDec18]],
+    // },
+    // {
+    //   target: pool.addresses.TINLAKE_CURRENCY,
+    //   call: ['balanceOf(address)(uint256)', address],
+    //   returns: [['poolCurrencyBalance', toDec18]],
+    // },
```
[the remark this answers is not in the document]
Comment (onnovisser) [reply centrifuge-app/src/utils/tinlake/useTInlakeInvestment.ts:93]: removed

File: centrifuge-app/src/utils/tinlake/useTinlakePermissions.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { useAddress } from '../useAddress'
+import { usePool } from '../usePools'
+import { Call, multicall } from './multicall'
+import { TinlakePool } from './useTinlakePools'
+
+export function useTinlakePermissions(poolId: string, address?: string) {
+  const addr = useAddress('evm') || address
+  const isTinlakePool = poolId.startsWith('0x')
+  const pool = usePool(poolId) as TinlakePool
+  return useQuery(['tinlakePermissions', poolId, addr], () => getPermissions(pool, address!), {
```
[the remark this answers is not in the document]
Comment (onnovisser) [reply centrifuge-app/src/utils/tinlake/useTinlakePermissions.ts:11]: fixed

File: centrifuge-app/src/utils/tinlake/useTinlakeBalances.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  pools.forEach((pool) => {
+    ;(['junior', 'senior'] as const).forEach((t, i) => {
+      const tranche = pool.tranches[i]
+      balances.tranches.push({
+        balance: new TokenBalance(multicallData.tokens[pool.id][t], tranche.currency.decimals),
+        currency: tranche.currency,
+        poolId: pool.id,
+        trancheId: tranche.id,
+      })
```
[the remark this answers is not in the document]
Comment (onnovisser) [reply centrifuge-app/src/utils/tinlake/useTinlakeBalances.ts:63]: renamed `t` to `trancheName`

File: centrifuge-app/src/utils/tinlake/useTinlakePools.ts (hunk 2 above)
Comment (onnovisser) [reply centrifuge-app/src/utils/tinlake/useTinlakePools.ts:496]: removed

File: centrifuge-app/src/pages/Pool/Overview/index.tsx [hunk 6]
```diff
@@ -296,10 +218,9 @@ export const PoolDetailOverview: React.FC<{
           <AnchorButton
             variant="secondary"
             icon={IconExternalLink}
-            href={`${import.meta.env.REACT_APP_SUBSCAN_URL}/account/${address}`}
+            href={explorer.address('')} // TODO: Add issuer address
```
[the remark this answers is not in the document]
Comment (onnovisser) [reply centrifuge-app/src/pages/Pool/Overview/index.tsx:221]: Made a comment on the AO Proxy ticket. will just leave the todo as well, just in case

**Pull request merged**